File: backend/models/pose.py
# backend/models/pose.py
"""
Pydantic models for pose history API responses.
"""
from __future__ import annotations

from typing import List
from pydantic import BaseModel


class PoseHistoryItem(BaseModel):
    """A single pose in the history."""
    t: float
    x: float
    y: float
    yaw: float


class PoseHistoryResponse(BaseModel):
    """Response containing the full pose history."""
    pose_history: List[PoseHistoryItem]


class PoseHistoryMeta(BaseModel):
    """Metadata about the pose history."""
    count: int
    t0: float
    t1: float

File: frontend/src/types/pose.ts
// frontend/src/types/pose.ts

export interface PoseHistoryItem {
    t: number;
    x: number;
    y: number;
    yaw: number;
}

export interface PoseHistoryResponse {
    pose_history: PoseHistoryItem[];
}

export interface PoseHistoryMeta {
    count: number;
    t0: number;
    t1: number;
}

export interface Pose {
    x: number;
    y: number;
    yaw: number;
}

export interface MidpointResponse {
    t: number;
    x: number;
    y: number;
    yaw: number;
    pose_history_distance: number;
    nav_path_distance: number;
    total_distance: number;
    midpoint_distance: number;
}
